nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -230,6 +230,19 @@ public void checkContractNegativeCases() {
         .doTest();
   }
 
+  @Test
+  public void checkCustomContractCases() {
+    makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:CustomContractAnnotations=com.uber.nullaway.testdata.CustomContract",
+                "-XepOpt:NullAway:CheckContracts=true"))
+        .addSourceFile("CheckCustomContractCases.java")
```

Comment:
For these smaller/self-contained tests, we are actually mostly using the `.addSourceLines(...)` method here. See e.g. `externalInitSupport`. Would you mind using that format and moving this either to the end of the test suite or - even better - to wherever the other tests about `@Contract` are down there? (We need to refactor this file, badly, but that's way out of scope for this PR so, for now, just want to keep it consistent with the other test cases :) )

Reply:
You will probably also want a third class here, were you check that the method invocations to `foo2` successfully pick up the contract information. E.g. that `foo(o).toString();` is safe when `o` is `@NonNull` and reports an error when `o` is `@Nullable`.